src/js/components/Box/StyledBox.js

```diff
@@ -289,7 +289,7 @@ const gapStyle = (directionProp, gap, responsive, border, theme) => {
     const responsiveBorderOffset =
       responsiveBorderMetric &&
       `${
-        parseMetricToNum(responsiveMetric) / 2 -
+        parseMetricToNum(responsiveMetric || metric) / 2 -
         parseMetricToNum(responsiveBorderMetric) / 2
       }px`;
```

Comment:
Unfortunately this will break spacing if any non-px metric is passed, as `'px'` is forcefully appended to the parsed number. But this is a more basic flaw in the border between implementation. I wonder if it makes sense to address it separately?

Reply:
@shb Im not sure what you mean by passing a non-px value? Are you referring to the t-shirt sizing we have like `small` `medium`?

Reply:
@shb I see the issue sorry it didn't click earlier so if someone passes `5em` the `borderOffSet` will not be correct because it is assuming `px` values. Alot of the math would also not be correct considering the `borderMetric` used to calculate the `borderOffSet` is in `px` Im thinking this may take some time in converting some of the values. Im inclined to open a separate issue to look into all of the values that would need to change.

Reply:
Yeah I think we can handle as a separate issue